prov/efa/src/rxr/rxr_read.c

```diff
@@ -155,7 +155,13 @@ struct rxr_read_entry *rxr_read_alloc_entry(struct rxr_ep *ep, int entry_type, v
 		}
 	} else {
 		assert(lower_ep_type == SHM_EP);
-		memset(read_entry->mr, 0, read_entry->iov_count * sizeof(struct fid_mr *));
+		/*
+		 * If CMA is NOT available, FI_MR_VIRT_ADDR will be cleared in shm provider,
+		 * use 0-based offset for shm's RMA operation.
+		 */
+		for (i = 0; i < read_entry->rma_iov_count; i++)
+			read_entry->rma_iov[i].addr = (efa_cma_cap) ?
+						      read_entry->rma_iov[i].addr : 0;
```

Comment:
Why not
```
if (!efa_cma_cap) {
    for (I=0; I < read_entry->rma_iov_count; ++i)
       read_entry->rma_iov[i].addr = 0;
}
```
and the setting of rma_iov[I] should be earlier. This place is for setting local mr. 

you can register local mr here instead of in ```rxr_inline_mr_reg```

Reply:
1. I can do the rma_iov[i] setting in function rxr_pkt_proc_shm_long_msg_rts() before calling rxr_read_alloc_entry(). 
2. For long message via shm, we need to do iov registration on the initiator side (initiator's registered `iov` will be sent to remote as `rma_iov`). I don't think the logic I added in rxr_inline_mr_reg can be done here (remote side). Please let me know if I miss something.